hw/ip/otbn/util/shared/information_flow.py

```diff
@@ -0,0 +1,557 @@
+        raise NotImplementedError()
+
+    def __repr__(self) -> str:
+        return 'InsnInformationFlowTest: {}'.format(self.as_string)
+
+
+class EqTest(InsnInformationFlowTest):
+    '''Tests that the operand is equal to the value.'''
+    def check(self, op_vals: Dict[str, int]) -> bool:
+        return op_vals[self.operand] == self.value
+
+
+class NotEqTest(InsnInformationFlowTest):
+    '''Tests that the operand is not equal to the value.'''
+    def check(self, op_vals: Dict[str, int]) -> bool:
+        return not (op_vals[self.operand] == self.value)
+
+
+class GeqTest(InsnInformationFlowTest):
+    '''Tests that the operand is greater than or equal to the value.'''
+    def check(self, op_vals: Dict[str, int]) -> bool:
+        return op_vals[self.operand] >= self.value
+
+
+class LeqTest(InsnInformationFlowTest):
+    '''Tests that the operand is less than or equal to the value.'''
+    def check(self, op_vals: Dict[str, int]) -> bool:
+        return op_vals[self.operand] <= self.value
+
+
+class MultiTest(InsnInformationFlowTest):
+    '''Combination of multiple tests.'''
+    def __init__(self, tests: List[InsnInformationFlowTest]):
+        self.tests = tests
+
+    def check(self, op_vals: Dict[str, int]) -> bool:
+        return all(t.check(op_vals) for t in self.tests)
+
+
+def _parse_iflow_test(test_yml: object, what: str,
+                      operands: List[Operand]) -> InsnInformationFlowTest:
+    '''Parses an item in the "test" list of a YAML information-flow rule.
+
+    Tests are expected to be strings of the form "<operand> <comparison>
+    <value>", where:
+      - <operand> is the name of one of the instruction's operands
+      - <comparison> is "==", "!=", ">=", or "<="
+      - <value> is an integer that is within the allowed range of values
+        for this operand (for instance, for a flag group it would be 0 or 1,
+        and for a register it would be between 0 and 31).
+
+    Returns a function that takes operand values as input and returns true if
+    the test passes.
+    '''
+    test = check_str(test_yml, what)
+    test = test.lower()
+    test_split = test.split(' ')
+    if len(test_split) != 3:
+        raise ValueError(
+            'Invalid information flow test format (expected "<operand> '
+            '<comparison> <value>"): got {} (for {})'.format(test, what))
+    opname, comparison, value_str = test_split
+
+    opnames = [op.name for op in operands]
+    if opname not in opnames:
+        raise ValueError(
+            'Invalid information flow test format for {}: operand {} not '
+            'found in operands list: {}'.format(what, opname, opnames))
+
+    if value_str.startswith('0x'):
+        # Interpret as hex.
+        value = int(value_str, 16)
+    elif value_str.isnumeric():
+        value = int(value_str)
+    else:
+        raise ValueError(
+            'Value {} in test {} for {} must be an integer.'.format(
+                value_str, test, what))
```

Comment:
I think you could do this more simply with just:
```suggestion
    try:
        value = int(value_str, 0)
    except ValueError:
        raise ValueError(
            'Value {} in test {} for {} must be an integer.'.format(
                value_str, test, what)) from None
```
(untested!)